provenance/repos.py

```diff
@@ -96,10 +97,34 @@ def name_set(artifact_set_or_id, name):
     return artifact_set.rename(name).put(repo)
 
 
+def transform_value(proxy_artifact, transformer_fn):
+    """
+    Transforms the underlying value of the `proxy_artifact` with
+    the provided `transformer_fn`. A new ArtifactProxy is returned
+    with the transformed value but with the original artifact.
+
+    The motivation behind this function is to allow archived files
+    to be loaded into memory and passed around while preserving
+    the provenance of the artifact. It could be used in any other
+    situation where you want a different representation of an
+    artifact value while allowing the provenance to be tracked.
+
+    Care should be taken when using this function however because
+    it will prevent you from reproducing exact artifacts from
+    the lineage since this transformer_fn will not be tracked.
```

Comment:
I think it is most likely more confusing to most people who would be users of the library. I did realize though that defining a transform function on the `Proxy` class could potentially block users from calling `transform` on the value that is being proxied. Isn't this a common scikit-learn method?